src/engraving/rendering/dev/measurelayout.cpp

```diff
@@ -855,14 +858,6 @@ void MeasureLayout::layoutMeasure(MeasureBase* currentMB, LayoutContext& ctx)
 
     measure->moveTicks(ctx.state().tick() - measure->tick());
 
-    if (ctx.conf().isLinearMode() && (measure->tick() < ctx.state().startTick() || measure->tick() > ctx.state().endTick())) {
-        // needed to reset segment widths if they can change after measure width is computed
-        //for (Segment& s : measure->segments())
-        //      s.createShapes();
-        ctx.mutState().setTick(ctx.state().tick() + measure->ticks());
-        return;
-    }
```

Comment:
can't see how this code could ever have done anything useful, and because of this early return the measure widths don't get correctly.

Reply:
Ok, unfortunately the early return *is* needed to prevent whole rests being misplaced (not centred). But there are other cases where it definitely breaks the layout.